qiskit/algorithms/eigen_solvers/numpy_eigen_solver.py

```diff
@@ -193,8 +193,67 @@ def _eval_aux_operators(
             values[key] = (value, 0.0)
         return values
 
+    @staticmethod
+    def _eval_transition_amplitudes(
+        aux_operators: ListOrDict[OperatorBase], wavefn, wavefm, threshold: float = 1e-12
+    ) -> ListOrDict[Tuple[complex, complex]]:
+        """Evaluate the transition amplitudes for the states n and m and the list of auxiliaries.
+
+        The formula we use here is a simpler version of Eq. 13 in
+        https://doi.org/10.1103%2Fphysrevresearch.3.023244. It is defined only for n!=m.
+        If n=m, then the _eval_aux_operators must be used instead.
+
+        """
+
+        values: ListOrDict[Tuple[complex, complex]]
+
+        # As a list, aux_operators can contain None operators for which None values are returned.
+        # As a dict, the None operators in aux_operators have been dropped in compute_eigenvalues.
+        if isinstance(aux_operators, list):
+            values = [None] * len(aux_operators)
+            key_op_iterator = enumerate(aux_operators)
+        else:
+            values = {}
+            key_op_iterator = aux_operators.items()
+        for key, operator in key_op_iterator:
+            if operator is None:
+                continue
+            value = 0.0
+            if operator.coeff != 0:
+                mat = operator.to_spmatrix()
+                # Terra doesn't support sparse yet, so do the matmul directly if so
+                # This is necessary for the particle_hole and other chemistry tests because the
+                # pauli conversions are 2^12th large and will OOM error if not sparse.
+                if isinstance(mat, scisparse.spmatrix) and not (
+                    isinstance(wavefn, StateFn) or isinstance(wavefm, StateFn)
+                ):
+                    value = mat.dot(wavefn).dot(np.conj(wavefm))
+                else:
+                    wavefnfm_plus = 1 / np.sqrt(2) * (wavefn + wavefm)
+                    wavefnfm_minus = 1 / np.sqrt(2) * (wavefn - wavefm)
+                    wavefnfm_i_plus = 1 / np.sqrt(2) * (wavefn + 1j * wavefm)
+                    wavefnfm_i_minus = 1 / np.sqrt(2) * (wavefn - 1j * wavefm)
+
+                    real_value = 0.5 * (
+                        StateFn(operator, is_measurement=True).eval(wavefnfm_plus)
```

Comment:
I suggest you construct `StateFn(operator, is_measurement=True)` only once, and then eval it four times

Reply:
Good point, thx